src/mockNetwork.ts

```diff
@@ -0,0 +1,96 @@
+import chalk from 'chalk'
+import { Response, Request } from './models'
+import { getRequestMatcher } from './requestMatcher'
+
+declare global {
+  interface Window {
+    fetch: jest.Mock
+  }
+}
+
+beforeEach(() => {
+  global.window.fetch = jest.fn()
+})
+
+afterEach(() => {
+  global.window.fetch.mockRestore()
+})
+
+const createDefaultResponse = async () => {
+  const response = {
+    json: () => Promise.resolve(),
+    status: 200,
+    ok: true,
+    headers: new Headers({ 'Content-Type': 'application/json' }),
+  }
+
+  return Promise.resolve(response)
+}
+
+const createResponse = async ({
+  responseBody,
+  status = 200,
+  headers,
+  delay,
+}: Response) => {
+  const response = {
+    json: () => Promise.resolve(responseBody),
+    status,
+    ok: status >= 200 && status <= 299,
+    headers: new Headers({ 'Content-Type': 'application/json', ...headers }),
+  }
```

Comment:
I think that making the destructuring inside the function is more legible. What do you think?

```suggestion
const createResponse = async (mockResponse: Response) => {
  const {
    responseBody,
    status = 200,
    headers,
    delay,
  } = mockResponse
  const response = {
    json: () => Promise.resolve(responseBody),
    status,
    ok: status >= 200 && status <= 299,
    headers: new Headers({ 'Content-Type': 'application/json', ...headers }),
  }
```